File: 백준/Silver/11399. ATM/ATM.py
# 1. 오름차순 기준으로 정렬

import sys

N = int(sys.stdin.readline())
arr = list(map(int ,sys.stdin.readline().split()))

# 정렬
arr.sort()

count = 0
result = 0

for i in range(N):
    count += arr[i]
    result += count
print(result)

File: 백준/Silver/11399. ATM/ATM.java

import java.io.*;
import java.util.*;

public class Main {

	public static void main(String[] args) throws IOException {
		BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

		int N = Integer.parseInt(bf.readLine());
		String[] sa = bf.readLine().split(" ");

		int[] arr = new int[N];
		for (int i = 0; i < N; i++) {
			arr[i] = Integer.parseInt(sa[i]);
		}

		Arrays.sort(arr);

		int sum = 0;
		int result = 0;

		for(int i=0; i<N; i++){
			sum += arr[i];
			result += sum;
		}

		System.out.println(result);

	}
}